### PR TITLE
Add getType() methods and merge createIndex() methods

### PR DESCRIPTION
**Pull request by SPalominos**
Add `get*Type*()` methods on `ITable` interface
Merge `createSpatialIndex` in `createIndex()`
Linked to the issues #219 and #217

### CODE REVIEW

File: data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ITable.java [hunk 1]
```diff
@@ -218,4 +222,70 @@ default boolean isEmpty() {
     default int[] getShape() {
```
Comment (ebocher) [inline data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ITable.java:222]: Doesn't understand this method

Comment (j3r3m1) [reply data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ITable.java:222]: I suppose return the "shape" of the table (a list containing the number of columns and the number of rows) ?

Comment (ebocher) [reply data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ITable.java:222]: Is it possible to find a more descriptive name ?

Comment (SPalominos) [reply data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ITable.java:222]: Yes, this method return the size of each dimension of the IDataSet. For an ITable, it return an array with the row and column count.
For the name, I'm open to any idea !

Comment (ebocher) [reply data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ITable.java:222]: getSize ?

Comment (SPalominos) [reply data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ITable.java:222]: ok

**Comment by SPalominos** [comment]
Ready to be merged

**Pull request merged**